API Products: Write new docs & revise naming (#5721)
Kong/docs.konghq.com: app/konnect/org-management/teams-and-roles/roles-reference.md

| Role                        | Description  |
|-----------------------------|--------------|
| Admin | Admin of an existing {{site.konnect_short_name}} service. The admins have all write access related to a gateway service and API product versions. |

> | Application Registration | Access to enable or disable application registration for a {{site.konnect_short_name}} service. |

Reviewer: ```suggestion
| Application Registration | Access to enable or disable application registration for an API Product. |
```

> | Creator | Access to create new {{site.konnect_short_name}} to productize services in the API product builder. The creator becomes the owner of the service they create, gaining admin access to the service. <br><br>This role does not provide access to creating sub-entities in a service such as API product versions, implementations, API specs, or plugins. See the `Service Admin`, `Maintainer`, or `Plugins Admin` roles. |

Reviewer: ```suggestion
| Creator | Access to create new API product to productize services in API Products. The creator becomes the owner of the API product they create, gaining admin access to the API product. <br><br>This role does not provide access to creating sub-entities in an API product such as API product versions. See the `API Product Admin` or `Maintainer` roles. |
```

| Role                        | Description  |
|-----------------------------|--------------|
| Admin | Admin of an existing {{site.konnect_short_name}} service. The admins have all write access related to a gateway service and API product versions. |
| Application Registration | Access to enable or disable application registration for an API Product. |
| Creator | Access to create new API product to productize services in API Products. The creator becomes the owner of the API product they create, gaining admin access to the API product. <br><br>This role does not provide access to creating sub-entities in an API product such as API product versions. See the `API Product Admin` or `Maintainer` roles. |
| Deployer | Access to implement and associate a {{site.konnect_short_name}} Gateway version to a runtime group. <br><br> Must also have the `Deployer` role for the associated runtime group. |

> | Maintainer | Access to read, edit, and deploy a {{site.konnect_short_name}} gateway service and its corresponding API product version versions, and manage its plugins. |

Reviewer 2: ```suggestion
| Maintainer | Access to read, edit, and deploy a Gateway service and its corresponding API product version versions, and manage its plugins. |
```
